File: scheduled/src/main/scala/com/stulsoft/pspring/scheduled/Application.scala
/*
 * Copyright (c) 2018. Yuriy Stul
 */

package com.stulsoft.pspring.scheduled

import com.typesafe.scalalogging.LazyLogging
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.{CommandLineRunner, SpringApplication}
import org.springframework.context.annotation.{Configuration, PropertySource}
import org.springframework.scheduling.annotation.{EnableScheduling, Scheduled}

/**
  * @author Yuriy Stul
  */
@SpringBootApplication(scanBasePackages = Array("com.stulsoft.pspring.scheduled"))
@PropertySource(Array("classpath:application.properties"))
@Configuration
@EnableScheduling
class Application extends CommandLineRunner with LazyLogging {
  @Autowired
  val someComponent: SomeComponent = null

  override def run(args: String*): Unit = {
    logger.info("==>run")

    logger.info("<==run")
  }

  @Scheduled(fixedDelay = 2000)
  def scheduleFixedDelay(): Unit = {
    logger.info("==>scheduleFixedDelay")
    someComponent.work()
    logger.info("<==scheduleFixedDelay")
  }
}

object Application extends App with LazyLogging {
  logger.info("==>Application")
  SpringApplication.run(classOf[Application], args: _*)
  logger.info("<==Application")
}

File: scheduled-java/src/main/java/com/stulsoft/pspring/scheduledj/Application.java
/*
 * Copyright (c) 2018. Yuriy Stul
 */

package com.stulsoft.pspring.scheduledj;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.CommandLineRunner;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.context.annotation.Conditional;
import org.springframework.context.annotation.PropertySource;
import org.springframework.scheduling.annotation.EnableScheduling;
import org.springframework.scheduling.annotation.Scheduled;

/**
 * @author Yuriy Stul
 */
@SpringBootApplication(scanBasePackages = {"com.stulsoft.pspring.scheduledj"})
@PropertySource({"classpath:application.properties"})
@ConditionalOnProperty(
        value = "app.scheduling.enable", havingValue = "true", matchIfMissing = true
)
@EnableScheduling
public class Application implements CommandLineRunner {
    private static final Logger logger = LoggerFactory.getLogger(Application.class);

    @Autowired
    private SomeComponent someComponent;

    @Override
    public void run(String... args) throws Exception {
        logger.info("==>run");

        logger.info("<==run");
    }

    @Scheduled(fixedDelay = 2000)
    public void scheduleFixedDelay() {
        logger.info("==>scheduleFixedDelay");
        someComponent.work();
        logger.info("<==scheduleFixedDelay");
    }
}
